File: Assignment_1/fourargs.cpp
// Copyright 2017 Nick VanDeusen nv314961@bu.edu

#include <iostream>
using namespace std;

int main(int argc, char* argv[]){
	// argc: number of arguments present
	// argv[]: character array of the arguments

	// if argc < 5, end for loop at argc rather than '5'
	/* we are ommitting argument 0, so argc <= 5 means
	   the user has entered 4 or less arguments. */
	if(argc <= 5){
		// send first x arguments through stdout (where x is 1-4)
		for(int i=1; i<argc; i++){
			cout << argv[i] << endl;
		}
	}
	else{
		// send first 4 arguments through stdout...
		for(int i=1; i<5; i++){
			cout << argv[i] << endl;
		}
		// ... and the remaining through stderr
		for(int i=5; i<argc; i++){
			cerr << argv[i] << endl;
		}
	}

	return 0;
}

File: Assignment_1/fourargs.py
# Copyright 2017 Nick VanDeusen nv314961@bu.edu

#!/usr/bin/python

import sys

# sys.argv: a list of present arguments
# len(sys.argv): # of arguments

if len(sys.argv) <= 5:
	# send first x arguments through stdout (where x is 1-4)
	for x in range(1,len(sys.argv)):
		print(sys.argv[x]);
else:
	# send first 4 arguments through stdout...
	for x in range(1, 5):
		print(sys.argv[x]);
	# ... and the remaining through stderr
	for x in range(5, len(sys.argv)):
		sys.stderr.write(sys.argv[x] +'\n');